src/sentry/dynamic_sampling/rules_generator.py

```diff
@@ -178,6 +203,18 @@ def generate_rules(project: Project) -> List[Union[BaseRule, ReleaseRule]]:
             enabled_biases = DynamicSamplingFeatureMultiplexer.get_enabled_user_biases(
                 project.get_option("sentry:dynamic_sampling_biases", None)
             )
+            # Key Transaction boost
+            if RuleType.BOOST_KEY_TRANSACTIONS_RULE.value in enabled_biases:
+                key_transactions = list(
+                    TeamKeyTransaction.objects.filter(
+                        organization_id=project.organization.id,
+                        project_team__project_id=project.id,
+                    )
+                    .values_list("transaction", flat=True)
+                    .distinct()[:BOOSTED_KEY_TRANSACTION_LIMIT]
+                )
```

Comment:
I don't like having logic put inside here. In `generate_boost_release_rules` we also have logic to fetch data and I would prefer to keep implementations similar. In addition, this adds complexity to this function that is out of scope.

In the future I would suggest to rework the implementation as discussed privately.

Reply:
yeah i agree, will extract to utils this block.

Reply:
fixed @iambriccardo